src/plots/cartesian/axes.js

```diff
@@ -2370,40 +2495,75 @@ axes.drawLabels = function(gd, ax, opts) {
             if(autoangle) {
                 positionLabels(tickLabels, autoangle);
             }
-            ax._lastangle = autoangle;
         }
+
+        if(ax._tickAngles) {
+            ax._tickAngles[cls] = autoangle === null ?
+                (isNumeric(tickAngle) ? tickAngle : 0) :
+                autoangle;
+        }
+    }
+
+    if(ax._selections) {
+        ax._selections[cls] = tickLabels;
     }
 
     var done = Lib.syncOrAsync([allLabelsReady, fixLabelOverlaps]);
     if(done && done.then) gd._promises.push(done);
     return done;
 };
 
-axes.drawTitle = function(gd, ax) {
-    var fullLayout = gd._fullLayout;
-    var tickLabels = ax._tickLabels;
+/**
+ * Draw axis dividers
+ *
+ * @param {DOM element} gd
+ * @param {object} ax (full) axis object
+ *  - {string} _id
+ *  - {string} showdividers
+ *  - {number} dividerwidth
+ *  - {string} dividercolor
+ * @param {object} opts
+ * - {array of object} vals (calcTicks output-like)
+ * - {d3 selection} layer
+ * - {fn} path
+ * - {fn} transFn
+ */
+function drawDividers(gd, ax, opts) {
```

Comment:
Can we remove ticks that would overlap dividers? I guess if we had inside ticks they should still be there, but not outside ticks.

Reply:
hmm this would get complicated with mirrored ticks... dividers won't mirror, right? So ideally the mirrored ones would still appear, but since we draw all ticks for a given x value as a single path containing all the mirrors, hiding just the main one would require a different path when we need to exclude the main one.

Reply:
> Can we remove ticks that would overlap dividers?

Yeah, we could. I tried to "solve" the problem by placing dividers below all ticks, but removing outside / `tickon:'boundaries'` ticks would be better. I'll see what I can do.



> dividers won't mirror, right?

Correct, dividers don't mirror.

Reply:
See my attempt in https://github.com/plotly/plotly.js/compare/multicategory...multicategory-polishing

Reply:
In https://github.com/plotly/plotly.js/pull/3300/commits/5da38513fc5224eaa29b7bf101d921732fe64c8e

Reply:
with new test mock in https://github.com/plotly/plotly.js/pull/3300/commits/204b2e803bbc93bbaf2a909025d24df0b6f67fa9